src/commands/functions-config-export.ts

```diff
@@ -1,151 +1,143 @@
 
-    if (!confirmed) {
-      throw new FirebaseError("Command aborted!");
+    // Display config in interactive mode
+    if (!options.nonInteractive) {
+      logBullet(clc.bold("Configuration to be exported:"));
+      logWarning("This may contain sensitive data. Do not share this output.");
+      console.log("");
+      console.log(JSON.stringify(configJson, null, 2));
+      console.log("");
     }
-  }
-}
-
-async function promptForPrefix(errMsg: string): Promise<string> {
-  logWarning("The following configs keys could not be exported as environment variables:\n");
-  logWarning(errMsg);
-  return await input({
-    default: "CONFIG_",
-    message: "Enter a PREFIX to rename invalid environment variable keys:",
-  });
-}
 
-function fromEntries<V>(itr: Iterable<[string, V]>): Record<string, V> {
-  const obj: Record<string, V> = {};
-  for (const [k, v] of itr) {
-    obj[k] = v;
-  }
-  return obj;
-}
+    const defaultSecretName = "RUNTIME_CONFIG";
+    let secretName = options.secret as string;
+    if (!secretName) {
+      secretName = await input({
+        message: "What would you like to name the new secret for your configuration?",
+        default: defaultSecretName,
+        nonInteractive: options.nonInteractive,
+        force: options.force,
+      });
+    }
 
-export const command = new Command("functions:config:export")
-  .description("export environment config as environment variables in dotenv format")
-  .before(requirePermissions, [
-    "runtimeconfig.configs.list",
-    "runtimeconfig.configs.get",
-    "runtimeconfig.variables.list",
-    "runtimeconfig.variables.get",
-  ])
-  .before(requireConfig)
-  .before(requireInteractive)
-  .action(async (options: Options) => {
-    const config = normalizeAndValidate(options.config.src.functions)[0];
-    const configDir = resolveConfigDir(config);
-    if (!configDir) {
+    const key = await ensureValidKey(secretName, options);
+    await ensureSecret(projectId, key, options);
+
+    const secretValue = JSON.stringify(configJson, null, 2);
+
+    // Check size limit (64KB)
+    const sizeInBytes = Buffer.byteLength(secretValue, "utf8");
+    const maxSize = 64 * 1024; // 64KB
+    if (sizeInBytes > maxSize) {
       throw new FirebaseError(
-        "functions:config:export requires a local env directory. Set functions[].configDir in firebase.json when using remoteSource.",
+        `Configuration size (${sizeInBytes} bytes) exceeds the 64KB limit for JSON secrets. ` +
+          "Please reduce the size of your configuration or split it into multiple secrets.",
       );
     }
 
-    let pInfos = configExport.getProjectInfos(options);
-    checkReservedAliases(pInfos);
-
+    const secretVersion = await addVersion(projectId, key, secretValue);
+    console.log("");
+
+    logSuccess(`Created new secret version ${toSecretVersionResourceName(secretVersion)}`);
+    console.log("");
+    logBullet(clc.bold("To complete the migration, update your code:"));
```

Comment:
wow, gnarly. could a template string let you do this in a single call instead of having to break it out line by line?